impl/src/main/java/org/jboss/weld/bootstrap/events/configurator/BeanConfiguratorImpl.java

```diff
@@ -311,7 +311,12 @@ public BeanConfiguratorImpl(Class<?> defaultBeanClass, BeanDeploymentFinder bean
     }
 
     public Bean<T> complete() {
-        return new ImmutableBean<>(this);
+        ImmutableBean<T> result = new ImmutableBean<>(this);
+        if (createCallback == null) {
```

Comment:
I am aware that creating bean and only then throwing exception isn't the best way, but without having `ImmutableBean`, there is no way to access information like types of bean or qualifiers.